### PR TITLE
[Tizen/API] Add unittest skeleton for Tizen-CAPI (TCT candidate) 

### PR DESCRIPTION
**Pull request by myungjoo**
This is to be the basis of TCT (Tizen Compliance Test) in the future.
    
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: packaging/capi-nnstreamer.manifest [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<manifest>
```
Comment (leemgs) [inline packaging/capi-nnstreamer.manifest]: Looks good.

File: tizen-api/include/tizen-api.h [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_BUF_FREE_BY_NNSTREAMER			= 0, /**< Default. Application should not deallocate this buffer. NNStreamer will deallocate when the buffer is no more needed */
+  NNS_BUF_DO_NOT_FREE				= 1, /**< This buffer is not to be freed. */
+} nns_buf_policy;
+
+/**
+ * @brief Enumeration for nns pipeline state
+ * @detail Refer to https://gstreamer.freedesktop.org/documentation/design/states.html
```
Comment (leemgs) [inline tizen-api/include/tizen-api.h]: :1st_place_medal:

File: tizen-api/include/tizen-api.h [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+  NNS_ERROR_PIPELINE_FAIL			= TIZEN_ERROR_STREAMS_PIPE, /**< Cannot create Gstreamer pipeline. */
+} nns_error_e;
+
+/**
+ * @brief Enumeration for buffer deallocation policies.
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_BUF_FREE_BY_NNSTREAMER			= 0, /**< Default. Application should not deallocate this buffer. NNStreamer will deallocate when the buffer is no more needed */
```
Comment (leemgs) [inline tizen-api/include/tizen-api.h]: It seems that the length of the line is too long. :)

File: tizen-api/include/tizen-api.h [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+ *          [  READY  ]                                            |
+ *               | "nns_pipeline_construct' completes              | "nns_pipeline_destroy"
+ *               V                                                 |
+ *          [         ] ------------------------------------------>|
+ *          [  PAUSED ] <-------------------+                      |
+ *               | "nns_pipeline_start"     | "nns_pipeline_stop"  |
+ *               V                          |                      |
+ *          [ PLAYING ] --------------------+----------------------+
+ *
+ * @since_tizen 5.5
```
Comment (leemgs) [inline tizen-api/include/tizen-api.h]: Is **@since_tizen** is valid when doxygen document is generated?

Comment (myungjoo) [reply tizen-api/include/tizen-api.h]: It's Tizen-doxygen syntax.

File: tizen-api/include/tizen-api.h (hunk 3 above)
Comment (myungjoo) [reply tizen-api/include/tizen-api.h]: Please note that tizen-api.h is not in the scope of this PR. That part is already merged.

File: tests/tizen_capi/unittest_tizen_capi.cpp [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+/**
+ * @brief Main gtest
+ */
+int main (int argc, char **argv)
+{
+  testing::InitGoogleTest (&argc, argv);
+
+  return RUN_ALL_TESTS ();
```
Comment (wooksong) [inline tests/tizen_capi/unittest_tizen_capi.cpp]: LGTM!

**Pull request merged**